src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -274,13 +351,129 @@ internal static byte[][] SegmentedEncodeNamedBitList(byte[] bigEndianBytes, int
                 dataSegment
             };
         }
-        
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an octet string (byte array).
+        /// </summary>
+        /// <param name="data">The data to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOctetString(byte[] data)
+        {
+            Debug.Assert(data != null);
+
+            // Because this is not currently public API the data array is not being cloned.
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.OctetString }, 
+                EncodeLength(data.Length),
+                data,
+            };
+        }
+
+        /// <summary>
+        /// Encode the segments { tag, length, value } of an object identifier (Oid).
+        /// </summary>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeOid(Oid oid)
+        {
+            if (oid == null)
+                throw new ArgumentNullException("oid");
+
+            // All exceptions past this point should just be "CryptographicException", because that's
```

Comment:
Window's message (for an OID starting with "3.") is `ASN1 bad arguments to function call` (or for the empty-string OID `The parameter is incorrect.`).

I guess I can float Argument_InvalidOidValue: `The OID value was invalid.` up into the Encodings tests so that they can continue to build the tests for this internal class.   Doesn't quite match what Windows does, but is arguably better.